pkg/apis/experimental/types.go

```diff
@@ -495,11 +495,12 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
-	// TODO: Add the ability to specify load-balancer IP just like what Service has already done?
-	// A list of rules used to configure the Ingress.
-	// http://<host>:<port>/<path>?<searchpart> -> IngressBackend
-	// Where parts of the url conform to RFC 1738.
-	Rules []IngressRule `json:"rules"`
+	// A default backend capable of servicing requests that don't match any
+	// IngressHost. Currently every IngressSpec requires a default backend.
+	Backend *IngressBackend `json:"backend,omitempty"`
+	// A list of host rules used to configure the Ingress.
+	Hosts []IngressHost `json:"rules"`
```

Comment:
The name "rules" is less confusing IMO, because that field is really some match rules (host+IngressRule still = Rule). The yaml works out. You could argue that `Rules []IngressHost` is a little confusing, otherwise I'm just going to rename the struct field to Rules. 

IngressHost isn't exposed anywhere and I don't think it's a bad name, but could be convinced of a rename there too. We have enough general general sounding names here, at least IngressHost is specific about the host.